roaring/container_stash.go

```diff
@@ -197,81 +451,71 @@ func (c *Container) setRuns(runs []interval16) {
 		newRuns := *(*[]interval16)(unsafe.Pointer(&reflect.SliceHeader{Data: uintptr(unsafe.Pointer(&c.data[0])), Len: stashedRunSize, Cap: stashedRunSize}))
 		copy(newRuns, runs)
 		c.pointer, c.len, c.cap = (*uint16)(unsafe.Pointer(&c.data[0])), int32(len(runs)), stashedRunSize
-		c.mapped = false // this is no longer using a hypothetical mmapped input array
+		c.flags &^= flagMapped // this is no longer using a hypothetical mmapped input array
 		return
 	}
+	if doCopy {
+		r2 := make([]interval16, len(runs))
+		copy(r2, runs)
+		h = (*reflect.SliceHeader)(unsafe.Pointer(&r2))
+	}
 	c.pointer, c.len, c.cap = (*uint16)(unsafe.Pointer(h.Data)), int32(h.Len), int32(h.Cap)
 	runtime.KeepAlive(&runs)
 }
 
 // Update updates the container
-func (c *Container) Update(typ byte, n int32, mapped bool) {
-	c.typ = typ
-	c.n = n
-	c.mapped = mapped
+func (c *Container) Update(typ byte, n int32, mapped bool) *Container {
```

Comment:
can't change this w/o a 2.0

Reply:
Re-reading this: This one may be genuinely impossible to avoid, in the case where we call Update on a frozen container.

But thinking about it: I think that's only used from PutContainerValues, and in fact, we only ever call that from the initial roaring unmarshal. The problem is that PutContainerValues is just plain wrong in a lot of cases -- it's assuming it's safe to just overwrite those fields, so it produces inconsistent states, which are later fixed up when we read in the offsets and attach the mmapped data.